cirq/google/engine/engine_sampler.py

```diff
@@ -64,3 +64,34 @@ def run_sweep(
     @property
     def engine(self) -> 'cirq.google.Engine':
         return self._engine
+
+
+def get_engine_sampler(processor_id: str, gate_set_name: str,
```

Comment:
I think that instead of taking a gate set name and having the NAMED_GATESETS dictionary, we should just take the gate set directly. There's no reason to add another layer of translation.

Reply:
at the risk of angering @dstrain115 by invoking the following argument: it's more pythonic to expose enum's using strings, especially in a user-facing api. See e.g. many numpy functions. In this case there are several advantages

 - the gateset can be specified by command line arg and/or environment variable
 - you don't have to find the correct import to find the gateset objects
 - gate_set_name can live comfortably in a pandas dataframe or database table

Reply:
I thought we were discouraging (if not deprecating) `engine.run` as it does not follow the sampler api

Reply:
The sampler api isn't flexible enough IMO to replace `run`.  It requires that you stick any extra information about your run into the sampler constructor, so if you really do care about this information you will end up creating samplers all over the place.

Reply:
Matt gave some good reasons to make it a string, so I'm fine with that. But we should be consistent.

Reply:
the only good reason I can think of not to use a string is that it makes it less extensible, but since there are only a set number of blessed gate sets, this seems fine to me.